Add Delete Snapshot High Level REST API

danielmitterdorfer opened the pull request:
With this commit we add the delete snapshot API to the Java high level
REST client.

Relates #27205

elasticmachine commented:
Pinging @elastic/es-core-infra

danielmitterdorfer commented:
Marked as "WIP" as we should wait for the create snapshot API so we can leverage it in tests (see https://github.com/elastic/elasticsearch/pull/31215).

hub-cap commented:
@danielmitterdorfer you can use the low level rest API in the tests, and then change them once the high level API is done. This is what I did when implementing `GET /repositories` before `PUT`.

javanna commented on client/rest-high-level/src/main/java/org/elasticsearch/client/SnapshotClient.java:
```diff
@@ -161,4 +163,34 @@ public void verifyRepositoryAsync(VerifyRepositoryRequest verifyRepositoryReques
+     * @return the response
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public DeleteSnapshotResponse deleteSnapshot(DeleteSnapshotRequest deleteSnapshotRequest, RequestOptions options) throws IOException {
```
can you call these methods delete without the snapshot part? they will be called as client.snapshot().delete() then which is how they are defined in our SPEC

danielmitterdorfer replied:
Makes sense. I've changed it.

danielmitterdorfer commented:
@hub-cap I've implemented the tests now based on the low-level client as per your suggestion and fixed the doc issues. Can you please have another look?

hub-cap commented:
Doh that was supposed to be a :shipit: not a :ship:

danielmitterdorfer commented:
Both icons work for me. ;) Thanks for the review!

danielmitterdorfer commented:
Backported to 6.x in 71b3ac2.
